docs/content/sips/016-component-imports.md

```diff
@@ -0,0 +1,53 @@
+title = "SIP 016 - Spin Component Imports" 
+template = "main" 
+date = "2023-09-22T16:27:43Z"
+---
+
+Summary: A proposal to introduce component imports to the Spin 2.0 manifest.
+
+Owner: <brian.hardock@fermyon.com>
+
+Created: Sep 19, 2023
+
+## Background / Goals
+The Spin 2.0 framework will support executing preview2 WebAssembly components. These components will include `import` and `export` statements. This manifest proposal is to enable Spin developers the ability to expressly configure how to satisfy the imports of their components. Furthermore, developers should be enabled to use the exports of a component to satisfy the imports of another.
+
+## Proposal
+
+### Changes to Spin Manifest 
+To support component imports, a new section will be added to the Spin 2.0 manifest design, `[component.<component-id>.import.<import-id>]`. The following sections describe the various options for specifying how to satisfy imports.
+
+> NOTE: `import-id` refers to one of the several optional for naming imports as per the [Component Model Explainer](https://github.com/WebAssembly/component-model/blob/main/design/mvp/Explainer.md#import-and-export-definitions).
+
+### Reference by ID
+Imports can be satisfied by associating an import with a component ID which links to a component defined elsewhere in the manifest.
+
+```toml
+[component.foobar.import."foo"]
+source = "other"
```

Comment:
The latter case; as far as i'm aware there hasn't been any discussion around supporting dangling `[component]` entries (not associated with a trigger).

Reply:
This concerns me - If we're choosing to only support components that conform to the Spin world for simplicity sake, it would be good for us to have a solid plan of what it would take to support components that don't conform to that world (and instead only expose the export that satisfies the required import).

Reply:
The v2 manifest supports components without a trigger.